install.sh

```diff
@@ -3256,12 +3319,26 @@ parse_command_line_arguments() {
             SINGLE_REPO_INSTALL="yes"
         elif [ "$p" = "--skip-install-into-existing" ]; then
             SKIP_INSTALL_INTO_EXISTING="yes"
+        elif [ "$prev_p" = "--prefix" ] && (echo "$p" | grep -qvE '^\-\-.*'); then
+            # Allow user to pass prefered install prefix
+            INSTALL_DIR="$p"
+            # Try to see if the path is given with a tilde
+            TILDE_REPLACED=$(echo "$INSTALL_DIR" | awk 'gsub("~", "'"$HOME"'", $0)')
+            if [ -n "$TILDE_REPLACED" ]; then
+                INSTALL_DIR="$TILDE_REPLACED"
+            fi
+            if [ ! -d "$INSTALL_DIR" ]; then
```

Comment:
Do we actually need to fail here?
Most (or hopefully all) code that will use this does an `mkdir -p`, right?

Reply:
Fail as early as possible. Argument validation after parsing...

Reply:
I think it shouldn't fail.
If I want `./install.sh --prefix ~/some/new/path`, I'd expect that to work.

Reply:
Ah ok, that usecase. Sure.